var/spack/repos/builtin/packages/fenics/package.py

```diff
@@ -43,135 +61,105 @@ class Fenics(CMakePackage):
+    conflicts('+slepc4py', when='~python')
+    conflicts('+slepc4py', when='~slepc')
+
+    # Patches
+    # patch('petsc-3.7.patch', when='petsc@3.7:')
+    patch('petsc-version-detection.patch', when='@1.6.0')
+    patch('hdf5~cxx-detection.patch', when='@:1.6.0')
+
+    patch('header_fix.patch', when='@2019.1.0.post0')
+    patch('petsc_3_11.patch', when='@2018.1.0.post1')
+
+    # enable extension support for fenics package
+    extends('python', when='+python')
+
+    # fenics python package dependencies
+    for ver in dolfin_versions:
+        wver = '@' + ver
+        depends_on('py-fenics-fiat{0}'.format(wver), type=('build', 'run'), when=wver)
+        if(Version(ver) < Version('2018.1.0')):
+            depends_on('py-fenics-instant{0}'.format(wver), type=('build', 'run'), when=wver)
+        else:
+            depends_on('py-fenics-dijitso{0}'.format(wver), type=('build', 'run'), when=wver)
+        depends_on('py-fenics-ufl{0}'.format(wver), type=('build', 'run'), when=wver)
+        if ver in ['2019.1.0', '2017.2.0']:
+            wver = wver + '.post0'
+        depends_on('py-fenics-ffc{0}'.format(wver), type=('build', 'run'), when=wver)
+
+    # package dependencies
+    depends_on('python@3:', type=('build', 'run'), when='+python')
     depends_on('eigen@3.2.0:')
-    depends_on('boost+filesystem+program_options+system+iostreams+timer+regex+chrono')
+    depends_on('pkgconfig')
+    depends_on('zlib', when='+zlib')
+
+    for ver in dolfin_versions:
+        if Version(ver) == Version('2019.1.0'):
+            depends_on('boost+filesystem+program_options+system+iostreams+timer+regex+chrono')
+        else:
+            depends_on('boost+filesystem+program_options+system+iostreams+timer+regex+chrono@1.68.0')
 
     depends_on('mpi', when='+mpi')
-    # FIXME: next line fixes concretization with petsc
     depends_on('hdf5+hl+fortran', when='+hdf5+petsc')
     depends_on('hdf5+hl', when='+hdf5~petsc')
-    depends_on('parmetis@4.0.2:', when='+parmetis')
     depends_on('metis+real64', when='+parmetis')
+    depends_on('parmetis', when='+parmetis')
     depends_on('scotch~metis', when='+scotch~mpi')
     depends_on('scotch+mpi~metis', when='+scotch+mpi')
-    depends_on('petsc@3.4:', when='+petsc')
-    depends_on('slepc@3.4:', when='+slepc')
+    depends_on('petsc', when='+petsc')
+    depends_on('slepc', when='+slepc')
+    depends_on('py-petsc4py@3.6:', when='+petsc+python')
     depends_on('trilinos', when='+trilinos')
     depends_on('vtk', when='+vtk')
     depends_on('suite-sparse', when='+suite-sparse')
     depends_on('qt', when='+qt')
 
-    depends_on('py-ply', type=('build', 'run'))
-    depends_on('py-six', type=('build', 'run'))
-    depends_on('py-numpy', type=('build', 'run'))
-    depends_on('py-sympy', type=('build', 'run'))
-    depends_on('swig@3.0.3:', type=('build', 'run'))
-    depends_on('cmake@2.8.12:', type='build')
+    depends_on('py-pybind11', type=('build', 'run'))
```

Comment:
```suggestion
    depends_on('py-pybind11@2.2.4', type=('build', 'run'))
```
At least in the latest release, you'll have to see if older versions have different requirements.

Reply:
Checking the `dolfin` 2018 and 2019 versions, they seem to work with `py-pybind11@2.2.4`, hence I will amend the spec as suggested.